### PR TITLE
support python3 and ubuntu 20.04

### PR DESCRIPTION
**Pull request by lguohan**
Signed-off-by: Guohan Lu <gulv@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Need to move kvm testbed infrastructure to ubuntu 20.04 to support kvm save/restore, which is important to repro testbed issues.

#### How did you do it?
provision ubuntu 20.04 server and run test

#### How did you verify/test it?
Tested on both 18.04 and 20.04
Only tested the provision script. Run test will come with some other PRs.

### CODE REVIEW

File: ansible/roles/vm_set/library/kvm_port.py [hunk 1]
```diff
@@ -29,17 +29,17 @@ def main():
 
     try:
         output = subprocess.check_output(
-                "virsh domiflist %s" % vmname, 
-                env={"LIBVIRT_DEFAULT_URI": "qemu:///system"}, 
-                shell=True)
+                "virsh domiflist %s" % vmname,
+                env={"LIBVIRT_DEFAULT_URI": "qemu:///system"},
+                shell=True).decode('utf-8')
```
Comment (lguohan) [inline ansible/roles/vm_set/library/kvm_port.py:34]: https://stackoverflow.com/questions/53261140/python-3-re-module-cannot-use-a-string-pattern-on-a-bytes-like-object-mac-ch

File: ansible/roles/vm_set/library/kvm_port.py [hunk 2]
```diff
@@ -29,17 +29,17 @@ def main():
...
     except subprocess.CalledProcessError:
         module.fail_json(msg="failed to iflist dom %s" % vmname)
 
     mgmt_port = None
     fp_ports = []
 
     for l in output.split('\n'):
-        fds = re.split('\s+', l)
+        fds = re.split('\s+', l.lstrip())
```
Comment (lguohan) [inline ansible/roles/vm_set/library/kvm_port.py:42]: due to virsh output format change, there will be leading whitespaces, removing them.

File: ansible/roles/vm_set/library/ceos_network.py [hunk 3]
```diff
@@ -207,7 +207,7 @@ def cmd(cmdline):
 
         with open(cmd_debug_fname, 'a') as fp:
             pprint("OUTPUT: %s" % stdout, fp)
-        return stdout
+        return stdout.decode('utf-8')
```
Comment (lguohan) [inline ansible/roles/vm_set/library/ceos_network.py:210]: https://stackoverflow.com/questions/53261140/python-3-re-module-cannot-use-a-string-pattern-on-a-bytes-like-object-mac-ch

File: ansible/roles/vm_set/library/sonic_kickstart.py [hunk 4]
```diff
@@ -42,7 +42,7 @@ def __init__(self, port, debug):
         self.d = debug
         self.d.debug('Starting')
         self.tn = Telnet('127.0.0.1', port)
-        self.tn.write('\r\n')
+        self.tn.write(b"\r\n")
```
Comment (lguohan) [inline ansible/roles/vm_set/library/sonic_kickstart.py:45]: https://docs.python.org/3/library/telnetlib.html

File: ansible/roles/vm_set/library/sonic_kickstart.py [hunk 5]
```diff
@@ -62,9 +62,9 @@ def cleanup(self):
     def pair(self, action, wait_for, timeout=60):
         self.d.debug('output: %s' % action)
         self.d.debug('match: %s' % ",".join(wait_for))
-        self.tn.write("%s\n" % action)
+        self.tn.write(b"%s\n" % action.encode('ascii'))
         if wait_for is not None:
-            index, match, text = self.tn.expect(wait_for, timeout)
+            index, match, text = self.tn.expect([ x.encode('ascii') for x in wait_for ], timeout)
```
Comment (lguohan) [inline ansible/roles/vm_set/library/sonic_kickstart.py:67]: https://docs.python.org/3/library/telnetlib.html

File: ansible/roles/vm_set/library/vm_topology.py [hunk 6]
```diff
@@ -128,15 +128,15 @@ def init(self, vm_set_name, topo, vm_base, dut_fp_ports, ptf_exists=True):
                 self.vm_base_index = self.vm_names.index(vm_base)
             else:
                 raise Exception('VM_base "%s" should be presented in current vm_names: %s' % (vm_base, str(self.vm_names)))
-            for k, v in topo['VMs'].iteritems():
+            for k, v in topo['VMs'].items():
```
Comment (lguohan) [inline ansible/roles/vm_set/library/vm_topology.py:131]: python3 removed iteritems()

File: ansible/roles/vm_set/tasks/docker.yml [hunk 7]
```diff
@@ -24,6 +24,13 @@
   become: yes
   when: host_distribution_version.stdout == "18.04"
 
+- name: Add docker repository for 18.04
```
Comment (yxieca) [inline ansible/roles/vm_set/tasks/docker.yml:27]: Please fix the version to 20.04

**Pull request merged**